### PR TITLE
[codex] Address review feedback

### PR DESCRIPTION
**Pull request by Kreijstal**
## Summary

Addresses actionable review feedback from the recent merged PR audit:

- hardens guarded runtime allocation size calculations and restores page-size fallback behavior
- removes MSVC-incompatible `_Generic` atomic macros while preserving pointer interlocked operations
- validates Pascal preprocessor `SHL`/`SHR` shift counts
- fixes Copilot log parser thinking/tool-call grouping and removes the broken unused `_build_args` helper
- avoids releasing borrowed KgpcType references during ShortString parameter remapping
- fixes visible-symbol prioritization and zero-sized array sizing paths
- avoids silently emitting partial tuple initializer assignments when element conversion fails
- preserves `expected_stderr` in timeout/exception failure artifacts
- refreshes the `pp_pas_bootstrap` golden output for the checked-out FPC source

## Validation

- `ninja -C build`
- `python3 -m py_compile scripts/parse-copilot-log.py tests/do_not_run_me_directly_but_through_meson.py`
- `meson test -C build --no-rebuild "cparser unit tests" "pascal parser unit tests"`
- `meson test -C build --no-rebuild "Compiler tests"` (883 subtests passed)
- `meson test -C build --no-rebuild "FPC RTL tests"` (236 subtests passed)

Untracked local files were left out of the commit: `.claude/`, `skip3_err.txt`, `skip_time.txt`.

## Summary by Sourcery

Address runtime safety issues, Pascal semantic/codegen edge cases, Copilot log parsing, and test diagnostics based on recent review feedback.

Bug Fixes:
- Restore guarded allocation page-size fallback and add overflow checks in the KGPC runtime allocator.
- Replace MSVC-incompatible generic atomic macros with explicit helper functions while preserving atomic semantics across compilers.
- Fix file lock acquisition to reliably clean up stale PID lockfiles without unbounded recursion.
- Ensure Pascal tuple/array-style initializers are not partially emitted when element expression conversion fails.
- Correct constructor detection and handling for class methods so implicit Self allocation and receiver expressions behave as expected.
- Fix visible-symbol prioritization and zero-sized array size calculations in x86-64 code generation and sizeof handling.
- Validate Pascal preprocessor SHL/SHR shift counts and report errors on invalid shift amounts.
- Fix Windows strndup shim to respect the specified maximum length and avoid overrunning the input string.
- Ensure shortstring parameter remapping does not release borrowed KgpcType references.
- Preserve expected_stderr contents in timeout and exception test failure artifacts.
- Avoid truncation issues in shortstring assignment x86-64 codegen by using the full buffer size for snprintf.

Enhancements:
- Refactor atomic operations behind typed helper functions shared between FPC intrinsics and runtime APIs.
- Centralize Pascal method constructor classification in a shared helper used by semantic checks for record access and function calls.
- Simplify Copilot log parsing to correctly group tool calls following assistant thinking blocks.

Tests:
- Add a regression test for class field virtual method calls to ensure correct constructor behavior and virtual dispatch semantics.

### CODE REVIEW

File: scripts/parse-copilot-log.py [hunk 1]
```diff
@@ -276,6 +250,21 @@ def parse(lines: list[Line]) -> RootNode:
...
+                tool.start_line = tool_start
+                root.children.append(AgentNode(
+                    thinking=ThinkingNode(
+                        text="\n".join(parts).strip(),
+                        start_line=start, end_line=i),
+                    tool_calls=[tool],
+                    start_line=start, end_line=i + consumed,
+                ))
+                i += consumed
+                continue
```
[the remark this answers is not in the document]
Comment (Kreijstal) [reply scripts/parse-copilot-log.py:266]: I checked this path and I don't think this is a real issue. `parts` and `start` are local to the current `copilot:` block. When a following `function:` block is found, the parser appends one `AgentNode`, advances `i` by the consumed tool block, and immediately continues the outer loop. The next turn re-enters the block with a fresh `parts` list and `start` value.
